common/map_test.cpp

```diff
@@ -14,11 +14,24 @@
 
 #include "common/raw_hashtable_test_helpers.h"
 
+// Workaround for std::pair comparison deficiency in clang-16.
+#if defined(__clang_major__) && __clang_major__ <= 16
+namespace std {
+template <class T, class U, class V, class W>
+  requires(convertible_to<V, T> && convertible_to<W, U>)
+inline auto operator==(pair<T&, U&> lhs, pair<V, W> rhs) -> bool {
+  return lhs.first == static_cast<T>(rhs.first) &&
+         lhs.second == static_cast<U>(rhs.second);
+}
+}  // namespace std
+#endif
```

Comment:
Should this be keyed on libc++ versions rather than Clang versions?

This is ... a pretty rough workaround...

Reply:
Good point, using `_LIBCPP_VERSION` now